client/packages/common/src/plugins/hooks/usePluginEvents.ts

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+import { usePluginProvider } from '../components';
+
+export const usePluginEvents = () => {
```

Comment:
added this as I was having a lot of re-render issues; and found the need to use a selector function to get values out of zustand state clumsy. Am worried that people won't use it correctly and will have render issues

Reply:
The advantage is usually talked about in terms of ease of use. personally, I've found the need to selectively extract rather annoying to use.